substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -94,17 +94,246 @@ private PosixJavaNetSubstitutions() {
 @TargetClass(className = "java.net.PlainDatagramSocketImpl")
 @Platforms({Platform.LINUX.class, Platform.DARWIN.class})
 final class Target_java_net_PlainDatagramSocketImpl {
+    @Alias boolean connected;
 
     @Substitute
     @SuppressWarnings({"static-method", "unused"})
     protected void bind0(int lport, InetAddress laddr) throws SocketException {
-        throw VMError.unsupportedFeature("Unimplemented: java.net.PlainDatagramSocketImpl.bind0(int, InetAddress)");
+        //        jobject fdObj = (*env)->GetObjectField(env, this, pdsi_fdID);
+        //        int fd;
+        //        int len = 0;
+        CIntPointer len_Pointer = StackValue.get(CIntPointer.class);
+        //        SOCKADDR him;
+        Socket.sockaddr him = StackValue.get(JavaNetNetUtilMD.SOCKADDR_LEN());
+        //
+        FileDescriptor fdObj = Util_java_net_DatagramSocketImpl.as_Target_java_net_DatagramSocketImpl(this).fd;
+        //        if (IS_NULL(fdObj)) {
+        if (fdObj == null) {
+            throw new SocketException("Socket closed");
+        }
+        //            JNU_ThrowByName(env, JNU_JAVANETPKG "SocketException",
+        //                            "Socket closed");
+        //            return;
+        //        } else {
+        int fd = PosixUtils.getFD(fdObj);
+        //            fd = (*env)->GetIntField(env, fdObj, IO_fd_fdID);
+        //        }
+        //
+        //        if (IS_NULL(iaObj)) {
+        if (laddr == null) {
```

Comment:
Tempting as it is to change the names of the names of the formal parameters, if you do that I have to check the correspondence between the `iaObj` in the C code and the `laddr` in your Java code.  On the other hand, I do allow myself to change the declaration of `jobject iaObj` to `InetAddress iaObj`, since we are in (typed) Java code, and that's the type `iaObj` has to have for this code to work.